### PR TITLE
[feat]: [CI-6334]: Publish plugin metadata

### PR DESCRIPTION
**Pull request by shubham149**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40715)

### CODE REVIEW

File: 332-ci-manager/service/src/main/java/io/harness/ci/beans/PluginMetadata.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @JsonProperty("secret") boolean secret;
+    @JsonProperty("default") String defaultVal;
+    @JsonProperty("allowed_values") List<String> allowedValues;
+  }
+
+  @Data
+  @Builder
+  @NoArgsConstructor
+  @AllArgsConstructor
+  public static class Output {
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/beans/PluginMetadata.java:47]: ignore unknown

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 4 of 22 files at r1, 20 of 21 files at r3, 1 of 1 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40715#-:-NIpHwcp-lz0VAPUsz7F:bnfp4nl)* status: :shipit: complete! ✓

**Review by harshjain12 — DISMISSED** [review]
> Reviewed 5 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40715)* status: :shipit: complete! ✓

File: 332-ci-manager/contracts/src/main/java/io/harness/cimanager/plugin/api/PluginMetadataResource.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+   * lists the available plugins
+   *
+   * @param page Page number of navigation
+   * @param size Number of entries per page
+   * @param searchTerm Filter entries based on plugin name
+   */
+  @GET
+  @Hidden
+  @Path("/")
+  @ApiOperation(value = "List plugins", nickname = "listPlugins")
```
[the remark this answers is not in the document]
Comment (shubham149) [reply 332-ci-manager/contracts/src/main/java/io/harness/cimanager/plugin/api/PluginMetadataResource.java:77]: not required

**Review by shubham149 — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40715#-:-NIq7vmOD4Xu_qqFSAtu:bnfp4nl)* status: 1. unresolved discussions.

**Pull request merged**